code/game/objects/structures/stool_bed_chair_nest/wheelchair.dm

```diff
@@ -193,3 +193,33 @@
 		pulling = null
 		usr.pulledby = null
 	..()
+		
+/obj/item/wheelchair
+	name = "wheelchair"
+	desc = "A folded wheelchair that can be carried around."
+	icon = 'icons/obj/furniture.dmi'
+	icon_state = "wheelchair_folded"
+	item_state = "wheelchair"
+	w_class = ITEMSIZE_HUGE // Can't be put in backpacks. Oh well.
+
+/obj/item/wheelchair/attack_self(mob/user)
+		var/obj/structure/bed/chair/wheelchair/R = new /obj/structure/bed/chair/wheelchair(user.loc)
```

Comment:
Everything contained in this `attack_self` is pretty much identical in your other `MouseDrop` proc, so make a new proc called `/obj/structure/bed/hair/wheelchair/proc/collapse()` or something and put this code in there, then call it from both places.